src/prank_parallel.cpp

```diff
@@ -37,33 +39,44 @@ int main(int argc, char** argv){
 
 
 
-    omp_set_num_threads(num_threads);
-
     //PHASE 1 : Read graph from input file
-    //SP: Should we do parallel file read as well ?
+    cout<<"Loading graph data..."<<endl;
+
     clock_t tStart = clock();
     int num_pages = 0;
-
-    //SP: we can think of some other way to store this information
     unordered_map <int, Page> pages; //mapping page ID to its struct
-    unordered_map <int, int> lookup; //program assigned id mapped to original ID of the page coming from file
 
-    cout<<"Loading graph data..."<<endl;
+    ifstream cin(input_file);
+
+    int id, out;
+    while(cin>>id) {
+        cin >> out;
+        if(!pages.count(id)) {
```

Comment:
If you want to use contains, I (think I) can upgrade the compile standard to cpp 20. I chose cpp 17 arbitrarily.